spinta/types/type.py

```diff
@@ -16,27 +18,64 @@ class Type:
         'link': {'type': 'string'},
     }
 
+    def is_valid(self, value):
+        raise NotImplementedError
+
 
 class PrimaryKey(Type):
-    pass
+
+    def is_valid(self, value):
+        # XXX: implement `pk` validation
+        return True
 
 
 class Date(Type):
-    pass
 
+    DEFAULT_FMT = "%Y-%m-%d"
 
-class DateTime(Type):
-    pass
+    def is_valid(self, value, date_fmt=DEFAULT_FMT):
+        if isinstance(value, str):
+            try:
+                datetime.strptime(value, date_fmt)
+            except ValueError:
+                return False
+            else:
+                return True
+
+
+
+class DateTime(Date):
+
+    DEFAULT_FMT = "%Y-%m-%d %H:%M:%S"
 
 
 class String(Type):
     schema = {
         'enum': {'type': 'array'},
     }
 
+    def is_valid(self, value):
+        try:
+            str(value)
+        except Exception:
+            # will never be called??
+            return False
+        else:
+            return True
+
 
 class Integer(Type):
-    pass
+
+    def is_valid(self, value):
+        if isinstance(value, int):
+            return True
+        else:
+            try:
+                int(value)
```

Comment:
Maybe add two step validation, using `load` and `check`? `load` would make sure, that value is loaded from a primitive value to a python native value, and then `check` would assume, that all types are already correct, and you only need to check if value is correct.

You will have to convert values to python-native values anyway.

Also a third step could be needed `prepare`, before saving data to database, python native values must be converted to a database native types. But maybe that is for another pull request.

Reply:
Do you mean to have something like:

```
data = load(context, model, data)
check(context,model, data)
check(context, model, model.backend, data)
```

in here: https://github.com/atviriduomenys/spinta/pull/19/files#diff-11540b6ded55485b62d4f20cf795971cR95

Reply:
Yes, something like that. First data should be `load`ed, going down all the tree and then loaded data should be passed to `check`.

Reply:
Will refactor this with next PR.

Will merge as is (tests are passing).